runhouse/resources/hardware/cluster.py

```diff
@@ -1946,3 +1950,162 @@ def _folder_mv(
 
     def _folder_exists(self, path: Union[str, Path]):
         return self.client.folder_exists(path=path)
+
+    @staticmethod
+    def get_clusters_from_den(cluster_filters: dict):
```

Comment:
this one still feels odd to me being inside the `cluster` class. i would expect methods here to be specifc to a specific cluster object, not a retreival of multiple cluster objects. we could also keep it inthis file, just as a standalone func outside of the cluster class